File: download_gradle.js
const https = require('https');
const fs = require('fs');
const path = require('path');
const zlib = require('zlib');

const url = 'https://services.gradle.org/distributions/gradle-8.7-bin.zip';
const dest = 'D:\\gradle-8.7.zip';

console.log('Downloading Gradle 8.7...');

const file = fs.createWriteStream(dest);
https.get(url, (response) => {
    if (response.statusCode === 302 || response.statusCode === 301) {
        console.log('Following redirect to:', response.headers.location);
        https.get(response.headers.location, (response2) => {
            response2.pipe(file);
            file.on('finish', () => {
                file.close();
                console.log('Download complete!');
                console.log('Extracting...');
                // Note: Node can't easily unzip, we'll use Java for that
            });
        });
    } else {
        response.pipe(file);
        file.on('finish', () => {
            file.close();
            console.log('Download complete!');
        });
    }
}).on('error', (err) => {
    fs.unlink(dest, () => {});
    console.error('Error:', err.message);
});

File: download_gradle.py
import os
import sys
import urllib.request
import zipfile
import shutil

# Download Gradle to user home directory
url = "https://services.gradle.org/distributions/gradle-8.7-bin.zip"
home = os.path.expanduser("~")
dest_zip = os.path.join(home, "gradle-8.7.zip")
dest_dir = os.path.join(home, "gradle-8.7")

print(f"Home: {home}")
print(f"Downloading {url}...")
print(f"To: {dest_zip}")

try:
    urllib.request.urlretrieve(url, dest_zip)
    print("Download complete!")
    
    print("Extracting...")
    with zipfile.ZipFile(dest_zip, 'r') as zip_ref:
        zip_ref.extractall(home)
    
    print("Running Gradle...")
    os.chdir(r"D:\Temp\REPassCard")
    gradle_bat = os.path.join(dest_dir, "bin", "gradle.bat")
    print(f"Gradle: {gradle_bat}")
    os.system(f'"{gradle_bat}" assembleDebug --no-daemon')
except Exception as e:
    print(f"Error: {e}")
